lessons/lesson-4/index.md

```diff
@@ -0,0 +1,226 @@
+# Lesson 3: Add in auth
+
+> In this lesson we will set up auth for the keystone admin UI
+
+## Where we left off
+
+We had added a second list to our keystone app for posts. We now have a keystone.js file that looks like:
+
+```js
+import { list } from '@keystone-next/keystone';
+import { text, timestamp, relationship } from '@keystone-next/keystone/fields';
+
+const lists = {
+  User: list({
+    fields: {
+      name: text({ isRequired: true }),
+      email: text({ isRequired: true }),
+      posts: relationship({ ref: 'Post.author', many: true }),
+    },
+  }),
+  Post: list({
+    fields: {
+      title: text(),
+      publishedAt: timestamp(),
+      status: select({
+        options: [
+          { label: 'Published', value: 'published' },
+          { label: 'Draft', value: 'draft' },
+        ],
+        defaultValue: 'draft',
+      }),
+      author: relationship({ ref: 'User.posts' }),
+      content: text({ ui: { displayMode: 'textarea' } }),
+    },
+  }),
+};
+
+export default {
+  db: {
+    provider: 'sqlite',
+    url: 'file:./keystone.db',
+  },
+  lists,
+};
+```
+
+## This Lesson
+
+This gives us a blog list that we can use. As we are looking at this list, we look at rounding out our keystone app, we want to put in auth, so that not everyone can view draft content, and not everyone can edit content.
+
+Keystone has very granular permissions controls, which you can read about [here](TK), however what we will focus on this tutorial is how to lock the Admin UI behind a password.
+
+## Adding The Password Field
+
+Keystone comes with an inbuilt password field, designed specifically for passwords. This takes care of hashing the password for storing in the database, as well as UI niceties such as making sure the password is obscured.
+
+We are going to use our existing `User` list for logging in, so we are going to add a new field to it:
+
+```js
+import { password } from '@keystone-next/keystone/fields';
+
+const lists = {
+  User: list({
+    fields: {
+      name: text({ isRequired: true }),
```

Comment:
Heads up that the latest release requires the following change across each field.

```suggestion
      name: text({ validation: { isRequired: true } } ),
```